daemon/container_operations.go

```diff
@@ -757,6 +757,10 @@ func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName
 
 	container.NetworkSettings.Ports = getPortMapInfo(sb)
 
+	if err := container.ToDisk(); err != nil {
+		return fmt.Errorf("Error saving container to disk: %v", err)
+	}
+
 	daemon.LogNetworkEventWithAttributes(n, "connect", map[string]string{"container": container.ID})
 	networkActions.WithValues("connect").UpdateSince(start)
```

Comment:
This metric is now measured after the `ToDisk()`. Not sure if that's important, just mentioning it